Detect correct Windows version for 8.1 and later.

MaartenBent opened the pull request:
This PR fixes the detection of Windows 8.1 and 10 by using  [RtlGetVersion](https://msdn.microsoft.com/en-us/library/windows/hardware/ff561910%28v=vs.85%29.aspx). This is broken since [GetVersionEx](https://msdn.microsoft.com/en-us/library/windows/desktop/ms724451%28v=vs.85%29.aspx) was deprecated (see [#15321](http://trac.wxwidgets.org/ticket/15321)).

The default Windows SDK does not include the required headers, so dynamic loading is necessary.
Looking at the source of  RTL_OSVERSIONINFO (winnt.h), only an unicode version is available.

Theoretically GetVersionEx could be replaced, as both functions are available since XP.
But to ensure compatibility with old operating systems, it is only used for Windows 8 (6.2) and later.

![Minimal Sample](http://i.imgur.com/GqYrbIS.png)

This PR could also be useful in `WX_3_0_BRANCH`.

TcT2k commented:
I think it's a acceptable workaround to use `RtlGetVerison` but it might discourage users from correctly including the [compatibility information](http://blogs.msdn.com/b/chuckw/archive/2013/09/10/manifest-madness.aspx) in their manifest file (With that info `GetVersionEx` and others give correct version numbers). Best solution would be if we could force that information with some kind of `#pragma` like the `commctrl` version in `defs.h`, but I haven't figured out how to do that (could be impossible).

MaartenBent commented:
IMO it is not up to wxWidgets to determine with which OS versions an application is compatible.
That said, if an application explicitly omits an OS from the compatibility information, it is still useful to know the actual OS (e.g. for debugging purposes).

Is there any reason why the manifests in `include\wx\msw` do not contain the compatibility info?

vadz commented:
Our manifest are not used by any modern MSVC version anyhow, so it's pretty immaterial what we do (or don't do) in them.

MaartenBent committed:
Fixed RtlGetVersion() usage.
On success it returns STATUS_SUCCESS, which equals 0.
The static function RawLoad() only returns a handle and does not initialize dllNtDll. The library is unloaded in the destructor.

See #15321

Closes #114

MaartenBent commented:
Hi, noticed your commit.
See updated commit to make it actually work.

vadz commented:
Oops, so much for quickly "porting" a (tested) snippet of code from another project to wxMSW without testing :-( Thanks for noticing and fixing these problems!
